cfme/tests/infrastructure/test_vm_clone.py

```diff
@@ -25,68 +22,39 @@
 ], scope="module")
 
 
-@pytest.yield_fixture(scope="function")
-def catalog_item(provider, vm_name, dialog, catalog, provisioning):
-    template, host, datastore, iso_file, catalog_item_type = map(provisioning.get,
-        ('template', 'host', 'datastore', 'iso_file', 'catalog_item_type'))
-
-    provisioning_data = {
-        'catalog': {'vm_name': vm_name,
-                    },
-        'environment': {'host_name': {'name': host},
-                        'datastore_name': {'name': datastore},
-                        },
-        'network': {},
-    }
-
-    if provider.type == 'rhevm':
-        provisioning_data['catalog']['provision_type'] = 'Native Clone'
-        provisioning_data['network']['vlan'] = provisioning['vlan']
-        catalog_item_type = "RHEV"
-    elif provider.type == 'virtualcenter':
-        provisioning_data['catalog']['provision_type'] = 'VMware'
-    item_name = fauxfactory.gen_alphanumeric()
-    catalog_item = CatalogItem(item_type=catalog_item_type, name=item_name,
-                  description="my catalog", display_in=True, catalog=catalog,
-                  dialog=dialog, catalog_name=template,
-                  provider=provider, prov_data=provisioning_data)
-    yield catalog_item
-
-
 @pytest.fixture(scope="function")
 def clone_vm_name():
     clone_vm_name = 'test_cloning_{}'.format(fauxfactory.gen_alphanumeric())
     return clone_vm_name
 
 
 @pytest.fixture
-def create_vm(appliance, provider, setup_provider, catalog_item, request):
-    vm_name = catalog_item.provisioning_data['catalog']["vm_name"]
-    catalog_item.create()
-    service_catalogs = ServiceCatalogs(appliance, catalog_item.name)
-    service_catalogs.order()
-    logger.info('Waiting for cfme provision request for service %s', catalog_item.name)
-    request_description = catalog_item.name
-    request_row = appliance.collections.requests.instantiate(request_description,
-                                                             partial_check=True)
-    request_row.wait_for_request()
-    assert request_row.is_succeeded()
-    return vm_name
+def create_vm(appliance, provider, setup_provider, request):
+    """Fixture to provision vm to the provider being tested"""
+    vm_name = 'test_clone_{}'.format(fauxfactory.gen_alphanumeric())
+    vm = VM.factory(vm_name, provider)
+    logger.info("provider_key: %s", provider.key)
+
+    @request.addfinalizer
+    def _cleanup():
+        vm.delete_from_provider()
+
+    if not provider.mgmt.does_vm_exist(vm.name):
+        logger.info("deploying %s on provider %s", vm.name, provider.key)
+        vm.create_on_provider(allow_skip="default", find_in_cfme=False)
+    return vm
 
 
 @pytest.mark.usefixtures("setup_provider")
-@pytest.mark.uncollectif(lambda: version.appliance_is_downstream())
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider))
```

Comment:
**REQUIRED**
```@pytest.mark.provider(VMwareProvider)```

Reply:
this led to 
```
cfme/tests/infrastructure/test_vm_clone.py
  cannot collect 'test_vm_clone' because it is not a function.

-- Docs: http://doc.pytest.org/en/latest/warnings.html
```

Reply:
```python
@pytest.mark.provider([VMwareProvider])
```